DEVELOPER_GUIDE.md

```diff
@@ -164,6 +164,21 @@ Run tests :
 ```
 ./gradlew clean build test
 ```
+## Generate Artifact
+
+In opensearch-sdk navigate to build/distributions. Look for tar ball in the form `opensearch-sdk-1.0.0-SNAPSHOT.tar`. If not found follow the below steps to create one:
+
+```
+./gradlew clean
+./gradlew build
+```
+Once the tar ball is generated navigate to /src/test/resources and look for extension.yml. Create one if not present
```

Comment:
Running the artifact would look for the extension.yml file within the `opensearch-sdk/build/distributions/opensearch-sdk-1.0.0-SNAPSHOT/src/test/resources/` instead of the `opensearch-sdk/src/test/resources` directory. This will cause the following error: 
`Exception in thread "main" java.io.FileNotFoundException: src/test/resources/extension.yml (No such file or directory)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:212)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:154)
        at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:354)
        at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:15)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3494)
        at org.opensearch.sdk.ExtensionsRunner.readExtensionSettings(ExtensionsRunner.java:90)
        at org.opensearch.sdk.ExtensionsRunner.<init>(ExtensionsRunner.java:67)
        at org.opensearch.sdk.ExtensionsRunner.main(ExtensionsRunner.java:386)`

Perhaps we should add this as a note and remind developers to create the extension.yml file within the correct directory prior to running the binary

Reply:
Create https://github.com/opensearch-project/opensearch-sdk/issues/86 to address this

Reply:
Let's resolve #86 before merging this then. @joshpalis @vibrantvarun WDYT?